### PR TITLE
[MP-48546] Make Miceplace using Workflowator

### PR DESCRIPTION
**Pull request by Arsen7**
https://miceportal.atlassian.net/browse/MP-48546

Here is how miceplace uses it: https://github.com/konvenit/miceplace/pull/1282

### CODE REVIEW

File: .github/workflows/compose_push.yaml [hunk 1]
```diff
@@ -37,45 +38,10 @@ jobs:
...
-
-        # check if all services are running
-        RUNNING=$(docker-compose -f docker-compose.test.yml ps --services --filter "status=running" | wc -l)
-        if [ "$RUNNING" != "$expected_services"]; then echo "not all services are running"; docker-compose -f docker-compose.test.yml logs; exit 1; fi
-
-        # check if the app is healthy
-        STATUS=$(docker inspect --format "{{.State.Health.Status}}" $(docker-compose -f docker-compose.test.yml ps -q ${{ inputs.application-name}}))
-        if [ "$STATUS" != "healthy" ]; then echo "${{ inputs.application-name }} is not healthy"; docker-compose -f docker-compose.test.yml ps; exit 1; fi
+    - name: Compose up
+      uses: konvenit/workflowator/.github/actions/compose@miceplace-test
```
Comment (Arsen7) [inline .github/workflows/compose_push.yaml]: This reference to the `miceplace-test` branch will be changed after this is merged. Or better - just before.

File: .github/actions/compose/action.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
+name: compose build and up
+description: Runs docker-compose up and waits for stuff to start properly
+
+inputs:
+  application-name:
+    required: true
+    type: string
+
+# TODO: maybe extract stuff to a script?
```
Comment (stereosupersonic) [inline .github/actions/compose/action.yml]: Do we need this todo?

File: .github/actions/compose/action.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+# An action sees the whole environment of the caller, but not the secrets.
+# If we want docker login here, we need to pass the user+pass as inputs
+# or as env.
+
+runs:
+  using: composite
+  steps:
+  - shell: bash
+    run: |
+      echo "::group::>>> Pull dependencies"
```
Comment (stereosupersonic) [inline .github/actions/compose/action.yml:14]: why so much noise with these grouping outputs or is this just debug code that can be removed ?

File: .github/actions/compose/action.yml (hunk 2 above)
Comment (Arsen7) [reply .github/actions/compose/action.yml]: No, we can live without it. Oops.. :sweat_smile:

**Review by stereosupersonic — COMMENTED** [review]
can we remove all of this **echo "::group::>>> XXX** and **echo "::endgroup::**

File: .github/workflows/compose_run.yaml [hunk 4]
```diff
@@ -79,9 +69,15 @@ jobs:
     - name: Run command with code coverage
       if: ${{ inputs.code_coverage }}
       run: |
+        # Could we download that file, review it, and then use that - approved by us - local copy?
         ci_env=`bash <(curl -s https://codecov.io/env)`
         docker-compose -f docker-compose.test.yml exec -T -e CI=true $ci_env ${{ inputs.application-name }} ${{ inputs.cmd}}
 
+    - name: Remove Algolia index (flaky!)
+      if: inputs.algolia_index && (success() || failure() || cancelled())
```
Comment (stereosupersonic) [inline .github/workflows/compose_run.yaml]: why is these conditionals "(success() || failure() || cancelled()" need? should this not always run if  algolia_index is set?

File: .github/workflows/compose_run.yaml [hunk 5]
```diff
@@ -36,40 +47,19 @@ jobs:
...
+      uses: konvenit/workflowator/.github/actions/compose@miceplace-test
+      with:
+        application-name: ${{ inputs.application-name }}
 
-        # check if the app is healty
-        STATUS=$(docker inspect --format "{{.State.Health.Status}}" $(docker-compose -f docker-compose.test.yml ps -q ${{ inputs.application-name }} ))
-        if [ "$STATUS" != "healthy" ]; then echo "${{ inputs.application-name }} is not healthy"; docker-compose -f docker-compose.test.yml ps; exit 1; fi
+    - name: Create Algolia index
+      if: ${{ inputs.algolia_index }}
+      run: |
```
Comment (stereosupersonic) [inline .github/workflows/compose_run.yaml:61]: the pipe is not needed if it's an one liner

File: .github/workflows/compose_run.yaml [hunk 6]
```diff
@@ -7,12 +7,17 @@ on:
         required: true
         type: string
       tag:
-        required: true
+        required: false
         type: string
+        default: ${{ github.event.pull_request.head.sha || github.sha }}
```
Comment (stereosupersonic) [inline .github/workflows/compose_run.yaml:12]: 👍

File: .github/actions/compose/action.yml (hunk 3 above)
Comment (Arsen7) [reply .github/actions/compose/action.yml:14]: Without these groups there is just a very very long wall of text, difficult to read and scroll.

I was experimenting with these "wrappable" groups, to make them less noisy in the code, but in the end this seems to be the best I could do for now:
obraz

File: .github/workflows/compose_run.yaml (hunk 4 above)
Comment (Arsen7) [reply .github/workflows/compose_run.yaml]: You're right. I think I was afraid of `always()` after reading this some time ago:
obraz

I still don't understand why the `always()` shall be avoided. :confused:

File: .github/workflows/compose_run.yaml (hunk 5 above)
Comment (Arsen7) [reply .github/workflows/compose_run.yaml:61]: True, but it's a pretty long line (so this way we can save 3 characters! :raised_hands: ), and with the pipe it's also consistent with how the next step below is formatted :point_down:

File: .github/actions/compose/action.yml (hunk 3 above)
Comment (Arsen7) [reply .github/actions/compose/action.yml:14]: I mean - these `::group::` thingies, when they appear in the output of the job, have a magic side effect of creating a wrappable section: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines

Comment (stereosupersonic) [reply .github/actions/compose/action.yml:14]: nice 👍  didn't know that

**Review by stereosupersonic — APPROVED** [review]

**Pull request merged**